src/BlazingPizza.Client/Pages/Checkout.razor

```diff
@@ -0,0 +1,65 @@
+@page "/checkout"
+@inject HttpClient HttpClient
+@inject OrderState OrderState
+@inject IUriHelper UriHelper
+@inject IJSRuntime JSRuntime
+
+<div class="main">
+    <AuthorizeView>
+        <NotAuthorized>
+            <h2>Signing in...</h2>
+        </NotAuthorized>
+        <Authorized>
+            <h2>Checkout</h2>
+
+            <div class="checkout-order-details">
+                <OrderReview Order="@OrderState.Order" />
+            </div>
+
+            <button class="checkout-button btn btn-warning" @onclick="@SubmitOrder" disabled="@isSubmitting">
+                Place order
+            </button>
+        </Authorized>
+    </AuthorizeView>
+</div>
+
+@functions {
+    bool isSubmitting;
+    [CascadingParameter] Task<AuthenticationState> AuthenticationStateTask { get; set; }
+
+    protected override async Task OnInitAsync()
+    {
+        var authState = await AuthenticationStateTask;
+        if (!authState.User.Identity.IsAuthenticated)
+        {
+            // The server won't accept orders from unauthenticated users, so avoid
+            // an error by making them log in at this point
+            await LocalStorage.SetAsync(JSRuntime, "currentorder", OrderState.Order);
```

Comment:
I'm assuming the idea here is that we need to preserve the state across a redirect? Let me know if that's not the case.

Reply:
Yes, exactly. In some ways doing a login pop-up would be simpler as it avoids this, but I think the redirect is more representative of a real-world solution.